tango_simlib/tests/test_validator.py

```diff
@@ -0,0 +1,358 @@
+    )
+    assert command_result in bi_directional_result
+    assert command_result in single_direction_result
+
+    # Same in both
+    assert "OtherCommand" not in bi_directional_result
+    assert "OtherCommand" not in single_direction_result
+
+    attr_result = (
+        "Attribute differs, [loggingLevelElement_A] specified but missing in device"
+    )
+    assert attr_result in bi_directional_result
+    assert attr_result in single_direction_result
+
+    attr_result = (
+        "Attribute differs, [loggingLevelElement_B] present"
+        " in device but not specified"
+    )
+    assert attr_result in bi_directional_result
+    assert attr_result not in single_direction_result
+
+    # Same in both
+    assert "OtherAttribute" not in bi_directional_result
+    assert "OtherAttribute" not in single_direction_result
+
+    prop_res = "Property [ControlModeDefault_A] differs, specified but missing in device"
+    assert prop_res in bi_directional_result
+    assert prop_res in single_direction_result
+
+    prop_res = (
+        "Property [ControlModeDefault_B] differs, present in device but not specified"
+    )
+    assert prop_res in bi_directional_result
+    assert prop_res not in single_direction_result
+
+    # Same in both
+    assert "AdminModeDefault" not in bi_directional_result
+    assert "AdminModeDefault" not in single_direction_result
+
+    spec_yaml = YAML_B
+    dev_yaml = YAML_A
+    bi_directional_result = compare_data(spec_yaml, dev_yaml, 1)
+
+    assert (
+        "lass differs, specified 'DishMaster_B', but device has 'DishMaster_A'"
+        in bi_directional_result
+    )
+    assert (
+        "Command differs, [ClearTaskHistory_B] specified but missing in device"
+        in bi_directional_result
+    )
+    assert (
+        "Command differs, [ClearTaskHistory_A] present in device but not specified"
+        in bi_directional_result
+    )
+    assert (
+        "Command [Capture] differs:\n\tdisp_level:"
+        "\n\t\tspecification: OPERATOR_B\n\t\tdevice: OPERATOR_A"
+    ) in bi_directional_result
+
+    # Same in both
+    assert "OtherCommand" not in bi_directional_result
+
+    assert (
+        "Attribute differs, [loggingLevelElement_B] specified but missing in device"
+    ) in bi_directional_result
+
+    assert (
+        (
+            "Attribute differs, [loggingLevelElement_A] present"
+            " in device but not specified"
+        )
+    ) in bi_directional_result
+
+    assert "OtherAttribute" not in bi_directional_result
+
+    assert (
+        ("Property [ControlModeDefault_B] differs, specified but missing in device")
+    ) in bi_directional_result
+
+    assert (
+        (
+            "Property [ControlModeDefault_A] differs, present in device but "
+            "not specified"
+        )
+    ) in bi_directional_result
+
+    # Same in both
+    assert "AdminModeDefault" not in bi_directional_result
+
+
+def test_empty_validation():
```

Comment:
I think we are missing a check for the classes in this test, or not?

Reply:
Covered in `assert not single_direction_result` since there should be no result.